### PR TITLE
feat(Commits): Remove APM logging and deduplicate results

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
A bug was reported that there are duplicate suggested commits/assignees. This is difficult to replicate in a local environment. Attempted to use logging in https://github.com/getsentry/sentry/pull/18526 but got reverted in https://github.com/getsentry/sentry/pull/18533 because the data structures were too large for our internal logging infrastructure.

Then tried logging through Sentry tracing instead in https://github.com/getsentry/sentry/pull/18548, but this is not the intended use case of Sentry APM because it samples requests.

So in this PR we are going to remove the Sentry APM logging and just deduplicate the results at the end of the request.

## Test Plan
Tested deduplication function in repl with the mock data from production issue. And wrote unit tests.

### CODE REVIEW

File: src/sentry/utils/committers.py [hunk 1]
```diff
@@ -254,7 +246,11 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
 
     for committer in committers:
         commit_ids = [commit.id for (commit, _) in committer["commits"]]
-        committer["commits"] = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
+        commits_result = [serialized_commits_by_id[commit_id] for commit_id in commit_ids]
```
Comment (markstory) [inline src/sentry/utils/committers.py:249]: If commits were indexed by the `key` would you be able to skip de-duplication? While we don't have constraints on commit.key, it *should* be uniqueish and is the 'id' we use when serializing commits.

Comment (NisanthanNanthakumar) [reply src/sentry/utils/committers.py:249]: @markstory when I checked on redash, there was only one commit, but we ended up with duplicates. I'm honestly not sure where and how the commits are being duplicated in the code. I suspect it within this function: https://github.com/getsentry/sentry/blob/c9cb310884029f685eec6b56dd8247616f83294a/src/sentry/utils/committers.py#L169 .

**Comment by scefali** [comment]
@NisanthanNanthakumar This change definitely needs unit tests

File: src/sentry/utils/committers.py [hunk 2]
```diff
@@ -254,7 +246,11 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
...
+        # Deduplicate commits
+        committer["commits"] = [
+            i for n, i in enumerate(commits_result) if i not in commits_result[n + 1 :]
```
Comment (scefali) [inline src/sentry/utils/committers.py:252]: @NisanthanNanthakumar Isn't the runtime of this going to be O(n^2)? How large is n here? I wonder if there is a way to do this more efficiently. Perhaps by building a set of already seen commits?.

File: src/sentry/utils/committers.py [hunk 3]
```diff
@@ -254,11 +246,24 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
...
         "feature.owners.has-committers",
         instance="hit" if committers else "miss",
         skip_internal=False,
     )
     return committers
+
+
+def dedupeCommits(commits):
+    seen = set()
+    result = []
```
Comment (scefali) [inline src/sentry/utils/committers.py:263]: @NisanthanNanthakumar No need to use two different data structures here. You can just use a dictionary and then return `dict.values()`

File: src/sentry/utils/committers.py [hunk 4]
```diff
@@ -254,11 +246,22 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
...
         instance="hit" if committers else "miss",
         skip_internal=False,
     )
     return committers
+
+
+def dedupeCommits(commits):
+    result = {}
+    for obj in commits:
+        if obj["id"] not in result:
```
Comment (scefali) [inline src/sentry/utils/committers.py:264]: @NisanthanNanthakumar Nit: you could have just done:
```
result = {}
for obj in commits:
  result[obj["id"]] = obj
```

**Review by scefali — APPROVED** [review]

File: src/sentry/utils/committers.py [hunk 5]
```diff
@@ -254,11 +246,22 @@ def get_serialized_event_file_committers(project, event, frame_limit=25):
...
 
     metrics.incr(
         "feature.owners.has-committers",
         instance="hit" if committers else "miss",
         skip_internal=False,
     )
     return committers
+
+
+def dedupeCommits(commits):
```
Comment (markstory) [inline src/sentry/utils/committers.py:261]: Function names in python are generally snake_case, so `dedupe_commits` here.

File: src/sentry/utils/committers.py (hunk 4 above)
Comment (joshuarli) [reply src/sentry/utils/committers.py:264]: This isn't really a nit, it's an objective improvement in readability. Going even further:

```
    return {
        c["id"]: c
        for c in commits
    }.values()
```

**Pull request merged**